### DISCUSSION TITLE
Auto-Matching for the Built-In Server

### DISCUSSION THREAD

**Discussion by glawleschkoff** [LeoKlaus/Paperparrot#205] [Feature Requests]
I know that auto-matching tags, correspondents or document types works if an external Paperless-ngx server is used. I like to have that functionality for the built-in server in the app. Thank You :D

**Comment by LeoKlaus** [comment]
Hey again Glawleshkoff,

I see, sorry for the misunderstanding. I don't really know how the auto matching in Paperless works so I have no idea how to implement something similar in Paperparrot.
If you have a suggestion for additional matching logic, feel free to add it here.

**Reply by glawleschkoff** [reply]
Okay, but don't you just use a version of Paperless-ngx as the built-in server? That should already come with that functionality.

**Reply by LeoKlaus** [reply]
Actually not, I've implemented all functionality myself.

**Reply by glawleschkoff** [reply]
Ah okay, I was thinking from the beginning that it's a version of Paperless-ngx under the hood as it appeared to me the easiest way to realize such an app. I mean, the app already has the functionality to connect to an external Paperless server. So there's the option to just host a Paperless-ngx server on the device itself and let your application use it's API. But I acknowledge that you have decided for another route and there could also be some problems when using multiple devices as you would have a Paperless-ngx server on every device with a common storage location (iCloud) and presumely Paperless-ngx is not intended to be used with multiple instances on the same database.
